pkg/deploy/controller/deployment/controller.go

```diff
@@ -59,6 +59,9 @@ func (c *DeploymentController) Handle(deployment *kapi.ReplicationController) er
 		// transition to failed immediately.
 		if deployutil.IsDeploymentCancelled(deployment) {
 			nextStatus = deployapi.DeploymentStatusFailed
+			if err := c.cancelDeployerPods(deployment); err != nil {
+				return err
```

Comment:
Maybe a larger question - when we cancel, why don't we update the RC and then delete the deployer pods?  Is that going to race with the deployment process pod controller?  Or are you just trying to ensure it happens before and after?